pkg/common/cluster_state.go

```diff
@@ -498,3 +514,32 @@ func (i *ClusterState) IsResourcesReady(cr *kc.Keycloak) (bool, error) {
 
 	return keycloakDeploymentReady && postgresqlDeploymentReady && keycloakRouteReady, nil
 }
+
+/*
+ *
+ * Migrate Backup Resources
+ *
+ */
+func (i *ClusterState) readKeycloakBackupCurrentState(context context.Context, cr *kc.Keycloak, controllerClient client.Client) error {
+	labelSelect := metav1.LabelSelector{
+		MatchLabels: cr.Labels,
+	}
+	backupCr := &v1alpha1.KeycloakBackup{}
+	backupCr.Namespace = cr.Namespace
+	backupCr.Name = model.MigrateBackupName + "-" + BackupTime
+	backupCr.Spec.InstanceSelector = &labelSelect
+
+	KeycloakBackup := model.KeycloakMigrationOneTimeBackup(backupCr)
+	KeycloakBackupSelector := model.KeycloakMigrationOneTimeBackupSelector(backupCr)
+
+	err := controllerClient.Get(context, KeycloakBackupSelector, KeycloakBackup)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			return err
+		}
+	} else {
+		i.KeycloakBackup = KeycloakBackup.DeepCopy()
+		cr.UpdateStatusSecondaryResources(i.KeycloakBackup.Kind, i.KeycloakBackup.Name)
```

Comment:
This is definitely not the right place to update the Secondary Resources. This should happen when creating the backup CR, not when reading it.

Reply:
My fault. I just copied the above function. CR KeycloakBackup is definitely not a secondary resource of CR Keycloak.